source/MaterialXCore/UnitConverter.cpp

```diff
@@ -0,0 +1,208 @@
+//
+// TM & (c) 2019 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <algorithm>
+#include <MaterialXCore/Util.h>
+#include <MaterialXCore/Value.h>
+#include <MaterialXCore/UnitConverter.h>
```

Comment:
The new source files UnitConverter.cpp and UnitConverter.h might be more at home in MaterialXGenShader, unless there's an important use case you're imagining for them in lower-level libraries.

Reply:
We want to handle 2 cases, 1 for shadergen and 1 for non-shadergen. For software rendering, as with colorspace, the desire is to be able to be able to get pre-transformed values without adding shader code for this.